### PR TITLE
Validate license key on reading cell values

### LINKED ISSUE #1181: Run license key validation on first read operation
### Description

To make integration with Handsontable easier, we don't want to display the `license key missing` warning when HyperFormula instance is created but only when user reads data from it.

Functions that should trigger license key validation:
- getCellValue
- getSheetValues
- getAllSheetsValues
- copy
- cut
- getRangeValues
- getNamedExpressionValue
- calculateFormula

Functions that must not trigger license key validation:
- buildFrom* methods
- updateConfig
- addNamedExpression
- addSheet
- setCellContents

## Links

https://github.com/handsontable/dev-handsontable/issues/587

### PR DESCRIPTION
**Pull request by sequba**
### Context

- added new event `CellValueRead`
- run license key validation on `CellValueRead` instead of on config update
- refactor code related to license key validation
- make spelling consistent `licence` -> `license`
- add unit tests for `CellValueRead` event and license key validation code
- remove `#LIC` error type - it will no longer be triggered

### How did you test your changes?

- [x] unit tests pass
- [x] manual test with different license keys (local HyperFormula build, demos from documentation)
- [ ] local docs build

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
Fixes #1181 

### Checklist:

- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [x] My code follows the code style of this project.
- [x] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: src/Emitter.ts [hunk 1]
```diff
@@ -314,6 +315,8 @@ export interface Listeners {
    * @category Batch
    */
   evaluationResumed: (changes: ExportedChange[]) => any,
+
+  cellValueRead: () => any,
```
Comment (sequba) [inline src/Emitter.ts:319]: Make this internal

**Review by adamu-handsontable — COMMENTED** [review]
I've asked the following questions:
- Should we not have kept LIC error -> we don't need it anymore, because it was only required for displaying it in cells which is no longer relevant as the trigger works only for read operations
- Question about resetting console.warn spy in beforeEach in `test/config.spec.ts` which could cause problem with resetting between tests -> it was changed to `afterEach`
- Asked if there is no possibility for `console.warn` to appear multiple times -> relevant tests were added to prevent it
- Asked for adding more tests which check each method that should throw a warning and each method which should not do that

File: src/Emitter.ts (hunk 1 above)
Comment (sequba) [reply src/Emitter.ts]: done

**Comment by sequba** [comment]
Closing. https://github.com/handsontable/hyperformula/issues/1181#issuecomment-1500024429

**Pull request closed without merging**